lib/ansible/modules/cloud/azure/azure_rm_webapp.py

```diff
@@ -0,0 +1,971 @@
+
+
+class AzureRMWebApps(AzureRMModuleBase):
+    """Configuration class for an Azure RM Web App resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str'
+            ),
+            plan=dict(
+                type='raw'
+            ),
+            frameworks=dict(
+                type='list'
+            ),
+            container_settings=dict(
+                type='dict',
+                options=container_settings_spec
+            ),
+            scm_type=dict(
+                type='str',
+            ),
+            deployment_source=dict(
+                type='dict',
+                options=deployment_source_spec
+            ),
+            startup_file=dict(
+                type='str'
+            ),
+            client_affinity_enabled=dict(
+                type='bool',
+                default=True
+            ),
+            dns_registration=dict(
+                type='bool'
+            ),
+            https_only=dict(
+                type='bool'
+            ),
+            skip_custom_domain_verification=dict(
+                type='bool'
+            ),
+            ttl_in_seconds=dict(
+                type='int'
+            ),
+            app_settings=dict(
+                type='dict'
+            ),
+            purge_app_settings=dict(
+                type='bool',
+                default=False
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        mutually_exclusive = [['container_settings', 'frameworks']]
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+
+        # update in create_or_update as parameters
+        self.client_affinity_enabled = True
+        self.dns_registration = None
+        self.skip_custom_domain_verification = None
+        self.ttl_in_seconds = None
+        self.https_only = None
+
+        self.tags = None
+
+        # site config, e.g app settings, ssl
+        self.site_config = dict()
+        self.app_settings = dict()
+        self.app_settings_strDic = None
+
+        # app service plan
+        self.plan = None
+
+        # siteSourceControl
+        self.deployment_source = dict()
+
+        # site, used at level creation, or update. e.g windows/linux, client_affinity etc first level args
+        self.site = None
+
+        # property for internal usage, not used for sdk
+        self.container_settings = None
+
+        self.purge_app_settings = False
+
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_webapp=None)
```

Comment:
Don't return info under `ansible_facts` unless they are facts about the host running the module. Anyway this info should really be returned under a _facts module. This module should just return info not already known by the user like `id`.

Reply:
resolved